mono/utils/mono-threads.h

```diff
@@ -159,6 +159,25 @@ enum {
 	ASYNC_SUSPEND_STATE_INDEX = 1,
 };
 
+typedef struct {
+#if G_BYTE_ORDER != G_BIG_ENDIAN
```

Comment:
Why do you care about endian? Does anything depend on the layout?

Reply:
The usual pattern here is:

typedef union _Foo {
  int AsInt; // or whatever
  struct {
    int field1 : 1;
    int field2: 4;
    // etc.
 };
} Foo;

You don't need two types or to name the struct in the union.

Reply:
Hm. Good point. I got rid of all the masking and shifting code, so the layout doesn't matter anymore.